packages/test/test-end-to-end-tests/src/test/loadModes.spec.ts

```diff
@@ -0,0 +1,375 @@
+
+		assert.notEqual(dataObject1.value, dataObject2.value, "counter values should not be equal");
+		assert.notEqual(
+			container1.deltaManager.lastSequenceNumber,
+			container2.deltaManager.lastSequenceNumber,
+			"container sequence numbers should not be equal",
+		);
+
+		assert.strictEqual(
+			sequenceNumber,
+			container2.deltaManager.lastSequenceNumber,
+			"container2 should still be at the specified sequence number",
+		);
+	});
+
+	describe("Expected error cases", () => {
+		it("Throw if sequence number not provided", async () => {
+			const headers: IRequestHeader = {
+				[LoaderHeader.loadMode]: {
+					opsBeforeReturn: "sequenceNumber",
+				},
+			};
+			try {
+				await loadContainer(container1.resolvedUrl, testDataObjectFactory, headers);
+				assert.fail("Did not throw expected error");
+			} catch (e: any) {
+				const expectedError = "sequenceNumber must be set to a non-negative integer";
+				assert.ok(e.message);
+				assert.strictEqual(e.message, expectedError, "Did not get expected error message");
+			}
+		});
+
+		it('Throw if sequence number is a negative integer"', async () => {
+			const headers: IRequestHeader = {
+				[LoaderHeader.loadMode]: {
+					opsBeforeReturn: "sequenceNumber",
+				},
+				[LoaderHeader.sequenceNumber]: -1,
+			};
+			try {
+				await loadContainer(container1.resolvedUrl, testDataObjectFactory, headers);
+				assert.fail("Did not throw expected error");
+			} catch (e: any) {
+				const expectedError = "sequenceNumber must be set to a non-negative integer";
+				assert.ok(e.message);
+				assert.strictEqual(e.message, expectedError, "Did not get expected error message");
+			}
+		});
+
+		it('Throw if opsBeforeReturn is not set to "sequenceNumber"', async () => {
+			const headers: IRequestHeader = {
+				[LoaderHeader.sequenceNumber]: 0, // Actual value doesn't matter
+			};
+			try {
+				await loadContainer(container1.resolvedUrl, testDataObjectFactory, headers);
+				assert.fail("Did not throw expected error");
+			} catch (e: any) {
+				const expectedError = 'opsBeforeReturn must be set to "sequenceNumber"';
+				assert.ok(e.message);
+				assert.strictEqual(e.message, expectedError, "Did not get expected error message");
+			}
+		});
+
+		it("Throw if attempting to pause at a sequence number before the latest summary", async () => {
+			const summarizer = await createSummarizerFromContainer(container1);
+			// Send 5 ops
+			const numIncrement = 5;
+			for (let i = 0; i < numIncrement; i++) {
+				dataObject1.increment();
+			}
+			await loaderContainerTracker.ensureSynchronized(container1);
+			const result = summarizer.summarizeOnDemand({ reason: "test" });
+			const submitResult = await result.summarySubmitted;
+			assert.ok(submitResult);
+
+			// 1s buffer to ensure we try to load from the summary
+			await new Promise((resolve) => setTimeout(resolve, 1000));
+
+			// Try to pause at sequence number 1 (before snapshot)
+			const sequenceNumber = 1;
```

Comment:
Nit: This should be fine, but maybe use 2 or 3 instead of 1 just to make it clear that it's not a boundary value test (there's nothing special about the first op)